scripts/build.sh

```diff
@@ -1,6 +1,33 @@
 #!/bin/bash
 set -e
 
+# Note:
+# CMAKE_BUILD_TYPE="Debug" adds "-O0 -g" flags by default
+# CMAKE_BUILD_TYPE="Release" adds "-O3 -DNDEBUG" by default
+if [[ "$BUILD_DEBUG" == "1" ]]; then
+    CMAKE_BUILD_TYPE="Debug"
+elif [[ "$BUILD_RELEASE" == "1" ]]; then
+    CMAKE_BUILD_TYPE="Release"
+elif [[ "$BUILD_PROFILING" == "1" ]]; then
+    CMAKE_BUILD_TYPE="Profiling"
+fi
+
+if [ $# -gt 0 ]; then
+    if [ "$1" == "Release" ] || [ "$1" == "--release" ]; then
+        CMAKE_BUILD_TYPE="Release"
+    elif [ "$1" == "Profiling" ] || [ "$1" == "--profiling" ]; then
+        CMAKE_BUILD_TYPE="Profiling"
+    elif [ "$1" == "Debug" ] || [ "$1" == "--debug" ]; then
+        CMAKE_BUILD_TYPE="Debug"
```

Comment:
Note that `[` is a synonym for the `test` program (frequently a shell built-in). POSIX does not specify that `test` accept a `==` operator; the string equality operator is `=`.

Since this is a bash script, we should probably prefer `[[` (which has nicer ergonomics anyway) which *does* include a `==` operator, but if we are going to use `[`, we should use `=` for string comparisons.

However, I might suggest using a case statement here. The above lines might look instead like so:

```bash
case "$1" in
    Release|--release) CMAKE_BUILD_TYPE='Release';;
    Profiling|--profiling) CMAKE_BUILD_TYPE='Profiling';;
    Debug|--debug) CMAKE_BUILD_TYPE='Debug';;
    *) printf 'Unrecognized build type…\nUsage: %s [Debug|Release|Profiling]\n' "$0";;
esac
```